### ISSUE TITLE
Gitian Build for High Sierra

### ISSUE THREAD

**Issue by jonspock** [PIVX-Project/PIVX#364]
currently if we use gitian to build for High Sierra, the executable crashes.  Native builds also using current autotools/make work but the dmg packages are not self contained. 2.3.1 gitian builds do not crash

**Comment by Fuzzbawls** [comment]
Having done some testing regarding this. It looks like compiling against Qt 5.9.x no longer results in an instant crash for High Sierra. This is using the exact same clang/llvm version that we currently use.

The downside, at least for the time being, is that simply updating the Qt version breaks determinism, so it is not yet a total solution.

**Comment by tofutim** [comment]
That is really interesting, if we do have to update Qt, does it get updated across all outputs?

**Comment by tofutim** [comment]
Fuzzbawls, I'm interested in taking a crack at this. Did you use the qt.mk in depends in a gitian build, or just use configure normally?

**Comment by Fuzzbawls** [comment]
@tofutim you're welcome to give it a go. Yes gitian builds use the `depends/packages/qt.mk` file and affects the version of Qt used in all outputs. I have not been able to track down the source of breaking determinism, but it is likely in a change that Qt made upstream to their internal build system.

**Comment by tofutim** [comment]
@Fuzzbawls  - any idea what I'm missing here?
```
home/ubuntu/build/destocoin/depends/x86_64-apple-darwin11/native -no-cups -no-egl -no-eglfs -no-freetype -no-gif -no-glib -no-icu -no-iconv -no-kms -no-linuxfb -no-libudev -no-mtdev -no-openvg -no-reduce-relocations -no-qml-debug -no-sql-db2 -no-sql-ibase -no-sql-oci -no-sql-tds -no-sql-mysql -no-sql-odbc -no-sql-psql -no-sql-sqlite -no-sql-sqlite2 -no-use-gold-linker -no-xinput2 -nomake examples -nomake tests -opensource -openssl-linked -optimized-qmake -pch -pkg-config -prefix /home/ubuntu/build/destocoin/depends/x86_64-apple-darwin11 -qt-libpng -qt-libjpeg -qt-pcre -system-zlib -reduce-exports -static -silent -v -no-feature-printer -no-feature-printdialog -release -xplatform macx-clang-linux -device-option MAC_SDK_PATH=/home/ubuntu/build/destocoin/depends/SDKs/MacOSX10.11.sdk -device-option MAC_SDK_VERSION=10.11 -device-option CROSS_COMPILE=x86_64-apple-darwin11- -device-option MAC_MIN_VERSION=10.11 -device-option MAC_TARGET=x86_64-apple-darwin11 -device-option MAC_LD64_VERSION=253.9
Project ERROR: Could not resolve SDK --show-sdk-path for 'macosx10.11'
```
# mac-qmake.conf
```
MAKEFILE_GENERATOR = UNIX
CONFIG += app_bundle incremental global_init_link_order lib_version_first plugin_no_soname absolute_library_soname
QMAKE_INCREMENTAL_STYLE = sublib
include(../common/macx.conf)
include(../common/gcc-base-mac.conf)
include(../common/clang.conf)
include(../common/clang-mac.conf)
QMAKE_MAC_SDK_PATH=$${MAC_SDK_PATH}
QMAKE_XCODE_VERSION=7.3.1
QMAKE_XCODE_DEVELOPER_PATH=/Developer
QMAKE_MACOSX_DEPLOYMENT_TARGET = $${MAC_MIN_VERSION}
QMAKE_MAC_SDK=macosx10.11
QMAKE_MAC_SDK.macosx.Path = $${MAC_SDK_PATH}
QMAKE_MAC_SDK.macosx.platform_name = macosx
QMAKE_MAC_SDK.macosx.SDKVersion = $${MAC_SDK_VERSION}
QMAKE_MAC_SDK.macosx.PlatformPath = /phony
!host_build: QMAKE_CFLAGS += -target $${MAC_TARGET}
!host_build: QMAKE_OBJECTIVE_CFLAGS += $$QMAKE_CFLAGS
!host_build: QMAKE_CXXFLAGS += $$QMAKE_CFLAGS
!host_build: QMAKE_LFLAGS += -target $${MAC_TARGET} -mlinker-version=$${MAC_LD64_VERSION}
QMAKE_AR = $${CROSS_COMPILE}ar cq
QMAKE_RANLIB=$${CROSS_COMPILE}ranlib
QMAKE_LIBTOOL=$${CROSS_COMPILE}libtool
QMAKE_INSTALL_NAME_TOOL=$${CROSS_COMPILE}install_name_tool
load(qt_config)
```
# darwin.mk
```
OSX_MIN_VERSION=10.9
OSX_SDK_VERSION=10.11
OSX_SDK=$(SDK_PATH)/MacOSX$(OSX_SDK_VERSION).sdk
LD64_VERSION=253.9
```

**Comment by tofutim** [comment]
so by breaking determinism you mean the hash of the output is always different

**Comment by Fuzzbawls** [comment]
correct

**Comment by tofutim** [comment]
how do i test if it breaks determinism or not?

**Comment by Fuzzbawls** [comment]
many iterations of gitian builds whilst comparing the resulting file shasums for every build

**Comment by Fuzzbawls** [comment]
Resolved in #754, closing
